### ISSUE TITLE
Creating cache from VQMod

### ISSUE THREAD

**Issue by jktoole** [vqmod/vqmod#51]
This isn't an issue with VQMod as such.

I'm working on a site that's hosting on Amazon's elastic cloud, so I can't use VQMod because the cloud doesn't like file_put_contents with the LOCK_EX flag. It's entirely possible that the cloud account could be configured to deal with that, but I don't have access to investigate/do anything about it.

The practical upshot of that is that I needed a way of "publishing" the VQmodded version of my OpenCart CE installation without including VQMod itself.

These hastily cobbled together scripts seem to do that (haven't run into any problems as yet), so I'm offering them up for discussion/disection/derision:

   https://gist.github.com/jktoole/be9fdee2263b48194d07

I should stress the 'hasty/cobbled together' part. E.g. one of the scripts is a Windows batch file that could probably be incorporated into the PHP by someone with more time/patience.

If this is is an inappropriate forum then please close the issue and I'm sorry for raising it - I didn't want to post it on the regular OpenCart forums in case someone copied/pasted/ran it without thinking and hosed their installation, then asked me for support :)

If there's enough interest, would be happy to spend more time on fine-tuning the scripts, adding options or handing them over to someone more qualified.

Ogun

**Comment by qphoria** [comment]
Couldn't you just remove EX_LOCK from the argument in vqmod.php ?

**Comment by jktoole** [comment]
That was the first thing I tried, but it caused problems when I tested it - some of which were probably the ones it was added to avoid.

The purpose of raising this issue was to see if there was any interest in cache generation, or in using VQMod to generate a modified copy of a site. The gist linked in the original post will (seems to anyway) generate a copy of a site with the vqmods applied.

VQMod generates cache on demand. In some situations (e.g. some cloud hosting environments), this isn't practical/feasible and pre-cached or pre-generated files are required.

This prototype has solved that problem in my specific situation. It's possible that it might be useful to other VQMod users.
